### PR TITLE
Navigation node as a link to an external page

### LINKED ISSUE #91: navigation node as a link to an external page
Introduce a static link node that only has the properties
* label
* category
* externalLinkUrl
* sameWindow 

If a node has a property externalLinkUrl, it is treated as a static link node and all irrelevant properties should be ignored. In the left-side navigation, these nodes should get a dedicated styleclass so they can be later distinguished visually.

### PR DESCRIPTION
**Pull request by parostatkiem**
**Description**

Changes proposed in this pull request:

- If a navigation node has `externalLinkURL` property defined, `pathSegment` property is ignored along with the whole routing mechanizm
- Some external links example in Angular app. One in top nav, others in side nav (`/#/ext`)
- Extra css classes are added to these nodes

Resolves #91

### CODE REVIEW

File: docs/navigation-configuration.md [hunk 1]
```diff
@@ -66,8 +66,8 @@ window.Luigi.setConfig({
 ## Nodes
 
 - **pathSegment** specifies the partial URL of the current segment. A static settings example reflects `luigidomain.test/settings`, while a dynamic one, prefixed with a colon, loads on any other value. **pathSegments** must not contain slashes.
-- **externalLinkUrl** specifies the external URL that Node leads to. If this property is set, **pathSegment** will be ignored.
-- **sameWindow** defines if the external URL should be opened in new or current tab. Works only if **externalLinkUrl** is set.
+- **externalLinkUrl** specifies the external URL that Node leads to. If this property is set, **pathSegment** is ignored.
+- **sameWindow** defines if the external URL sis opened in a new or current tab. Works only if **externalLinkUrl** is set.
```
Comment (klaudiagrz) [inline docs/navigation-configuration.md]: ~sis~ -> `is`

File: docs/navigation-configuration.md [hunk 2]
```diff
@@ -52,27 +52,32 @@ window.Luigi.setConfig({
...
-- **navigationContext** contains a named Node that is mainly for use in combination with a dynamic **pathSegment** to start navigation from a dynamic Node using ` LuigiClient.navigationManager().fromContext('contextname')`.
-- **context** sends the specified object as context to the view. Use this parameter in combination with the dynamic **pathSegment** to receive the context through the context listeners of **Luigi client**. This is an alternative to using the dynamic value in the **viewUrl**.
-- **defaultChildNode** sets the child node that Luigi activates automatically if the current node has no **viewUrl** defined. Provide **pathSegment** of the child node you want to activate as a string.
-- **isolateView** renders the view in a new frame when you enter and leave the Node. This setting overrides the same-domain frame re-usage. The **isolateView** is disabled by default.
+- **externalLink** is an object specifying that the node should link to an external URL. If this property is set, **pathSegment** is ignored. It consists of the following properties:
```
Comment (klaudiagrz) [inline docs/navigation-configuration.md]: `` **externalLink** is an object which specifies that the Node links to an external URL.``

File: docs/navigation-configuration.md [hunk 3]
```diff
@@ -52,27 +52,32 @@ window.Luigi.setConfig({
...
+  - **sameWindow** defines if the external URL is opened in a new or current tab.
+  - **url** the external URL that Node leads to.
```
Comment (klaudiagrz) [inline docs/navigation-configuration.md]: `` **url** is the external URL that the Node leads to.``

File: docs/navigation-configuration.md [hunk 4]
```diff
@@ -52,27 +52,32 @@ window.Luigi.setConfig({
...
+
+* **label** contains the display name of the navigation Node.
+* **hideFromNav** shows or hides a navigation Node. You can still navigate to the Node but it will not show up in the top or left pane.
+* **viewUrl** contains the URL or path to a view that renders when entering the navigation Node. Use either a full URL or a relative path. This value may consist of variables if you have specified a **navigationContext** with a dynamic **pathSegment**. If **viewUrl** is undefined, Luigi activates the child node specified in **defaultChildNode**. When both **viewUrl** and **defaultChildNode** are undefined, Luigi opens the first child of the current node.
```
Comment (klaudiagrz) [inline docs/navigation-configuration.md]: 1. ...renders `when you enter the navigation Node.`
2. ...`Luigi activates the child Node specified`... -> capitalize Node
3. `the current Node`

File: docs/navigation-configuration.md [hunk 5]
```diff
@@ -52,27 +52,32 @@ window.Luigi.setConfig({
...
+* **navigationContext** contains a named Node that is mainly for use in combination with a dynamic **pathSegment** to start navigation from a dynamic Node using `LuigiClient.navigationManager().fromContext('contextname')`.
+* **context** sends the specified object as context to the view. Use this parameter in combination with the dynamic **pathSegment** to receive the context through the context listeners of **Luigi client**. This is an alternative to using the dynamic value in the **viewUrl**.
```
Comment (klaudiagrz) [inline docs/navigation-configuration.md]: `This is an alternative to` ~using~ ``the dynamic value in`` ~the~ ``**viewUrl**.``

File: docs/navigation-configuration.md [hunk 6]
```diff
@@ -52,27 +52,32 @@ window.Luigi.setConfig({
...
+* **defaultChildNode** sets the child node that Luigi activates automatically if the current node has no **viewUrl** defined. Provide **pathSegment** of the child node you want to activate as a string.
```
Comment (klaudiagrz) [inline docs/navigation-configuration.md]: Please capitalize Node everywhere within this document

File: core/examples/luigi-sample-angular/src/app/overview/overview.component.ts [hunk 7]
```diff
@@ -31,6 +31,11 @@ export class OverviewComponent {
       link: '/projects/pr1/users/groups',
       text: 'dynamic Nodes',
       description: 'navigation Node configuration to set a specific node as dynamic'
+    },
+    {
+      link: 'http://google.com',
```
[the remark this answers is not in the document]
Comment (parostatkiem) [reply core/examples/luigi-sample-angular/src/app/overview/overview.component.ts]: I'd leave the external link in the top navigation to show the developer he can also put it there.

[the remark this answers is not in the document]
Comment (parostatkiem) [reply core/examples/luigi-sample-angular/src/app/overview/overview.component.ts]: What do you think @maxmarkus ?

**Comment by kwiatekus** [comment]
The external link should have dedicated style class so that user can distinguish them. This doesnt work yet

**Pull request merged**